tests/e2e/java_test.go

```diff
@@ -0,0 +1,60 @@
+package e2e
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const files = "examples/java/"
+
+var _ = Describe("odoJavaE2e", func() {
+	var t = "java"
+	var projName = fmt.Sprintf("odo-%s", t)
+
+	// Create a separate project for Java
+	Context("create java project", func() {
+		It("should create a new java project", func() {
+			session := runCmd("odo project create " + projName)
+			Expect(session).To(ContainSubstring(projName))
+		})
+	})
+
+	// Test Java
+	Context("odo component creation", func() {
+
+		It("Should be able to deploy a .jar file", func() {
+			runCmd("odo create wildfly jar-test --binary " + files + "wildwest-1.0.jar")
```

Comment:
can we download binary if possible ? just saying because it's increasing repo size

Reply:
If we do that, then each time the test is ran it'd waste bandwith :-1: 

What do you think @kadel ? Should we keep the binaries in the repo or download externally?

Reply:
+1 for maintaining inside our repo or atleast host them in a reliable trusted place.
We shouldn't ideally rely on external binaries in IMHO anyway I do not think Odo binary size would increase without addition of more go sources(from what I gathered from @kadel during our conversation around why it was a challenge for packaging config file as a separate non-go text file with our source with all default values...)

But we could as well use raw git url to fetch binaries from our own repo as well(If absolutely required)

Reply:
let's keep it in our repo, it will eliminate flakes due to timeouts etc..